gluten-core/src/main/java/io/glutenproject/substrait/expression/ListLiteralNode.java

```diff
@@ -33,13 +34,18 @@ protected void updateLiteralBuilder(Builder literalBuilder, ArrayData array) {
     Object[] elements = array.array();
     TypeNode elementType = ((ListNode) getTypeNode()).getNestedType();
 
-    Expression.Literal.List.Builder listBuilder = Expression.Literal.List.newBuilder();
-    for (Object element : elements) {
-      LiteralNode elementNode = ExpressionBuilder.makeLiteral(element, elementType);
-      Expression.Literal elementExpr = elementNode.getLiteral();
-      listBuilder.addValues(elementExpr);
+    if (elements.length > 0) {
+      Expression.Literal.List.Builder listBuilder = Expression.Literal.List.newBuilder();
+      for (Object element : elements) {
+        LiteralNode elementNode = ExpressionBuilder.makeLiteral(element, elementType);
+        Expression.Literal elementExpr = elementNode.getLiteral();
+        listBuilder.addValues(elementExpr);
+      }
+      literalBuilder.setList(listBuilder.build());
+    } else {
+      Type.List.Builder listTypeBuilder = Type.List.newBuilder();
+      listTypeBuilder.setType(elementType.toProtobuf());
```

Comment:
if if element type is NullType, we will still fallback it right ? e.g., `select array();`

Reply:
It's related to https://github.com/oap-project/gluten/pull/2996. It will not fallback once NullType is supported